### ISSUE TITLE
Get the value of the "search" input

### ISSUE THREAD

**Issue by RawiaAhmed** [softsimon/ngx-bootrap-multiselect#191]
I'm using the option enableSearch: true, is there any way I can get the value that I'm typing in the search input?

**Comment by nicholasguyett** [comment]
That value is not currently exposed to external code.  Its intended to be a UI enhancement rather than an actual value.

What exactly do you intend to do with the search value?

**Comment by RawiaAhmed** [comment]
I wanted to call a service with it's value, so that it won't only filter the existing options it also is used in getting new data with the new typed value.

**Comment by nicholasguyett** [comment]
Okay.  So what you need is to be able to reference dynamically loaded data which represents a portion of the full data set?  In other words, you want the search to pick from a full set of options but there is more data than you can load onto the page at one time?

If that's correct, I can see why you'd want that.  On the one hand, that seems a reasonable desire, but on the other, I'm wondering if maybe you should be using a different component for this.

Once it pulls in the new data, do you want that new data to be included in the full list when the user clears their search option or should still only show the initial data?  And how do you intend to communicate to the user that they have more options to pick from than just what's in the initial view?  I know I'd be surprised to find out I had to type to get additional choices.

**Comment by RawiaAhmed** [comment]
You are right that's what I want, and once it pulls new data it will still be there.
The whole idea now is that I want the filtering to be done also on the back-end side not just the current data, what I thought of is to get the data by using javascript selectors.

**Comment by tmurphyHOM** [comment]
![multiselectwithpagedresults](https://user-images.githubusercontent.com/19195799/26882862-d3ad3634-4b58-11e7-9957-915148900549.JPG)

I needed the same type of implementation:
-  paged load of the drop down, 
- display  # of total found along with display count, 
- display  the search box only when the # displayed is less than total count (still demoing this so when visibility of the search box is triggered may change) 
- and  providing a visual for "run filter" (or go get paged data based on what I just entered in the search box".)  
-  I opted to leave in the current search within
This component offered 95% of what I needed, so I just pulled down the code and have been softly modifying it. Most of the code required really was outside this component - initializing the contents and correctly setting the setting values based on data returned from the server.  I reload the data on successful retrieval of the filtered data.   

Still validating that this flow (specifically the run filter) will be intuitive and provide everything we need, but right now I am happy with it.

To make this work I made the following minor changes to the component:
- added 2 new settings:  totalItems and itemsPerPage:  used these values to determine the setting for enableSearch (true/false) 
- added a button to the search box that emits a new event - the event that sends the search box contents to the server and returns an updated list.  

I have attached a screen shot of the current rendering.

**Comment by nicholasguyett** [comment]
@tmurphyHOM, have you made a pull request?

**Comment by tmurphyHOM** [comment]
No, I have not.  I was initially concerned my changes would be too custom - so far that is not the case.  I'll work on the pull later today.

**Comment by tmurphyHOM** [comment]
Wanted to give you an update on this.  I created a fork for this here:  https://github.com/tmurphyHOM/angular-2-dropdown-multiselect and a playground for it here:  https://github.com/tmurphyHOM/paging-edits.  
As i started pulling this out of my implementation, I found a few things you might need that my implementation did not need and added those in.  My implementation is to support a filter bar on a list.  I only have a few rare instances where i have more than 30 items in the drop down so for those instances, I needed a multi-select dropdown to use in that filter bar that supported paging.  Most of the paging work is performed external to the drop down component - so I created the paging-edits playground that might help you work through this and to see if this is actually something that would work for you.
